documentation/CLI_usage.md

```diff
@@ -0,0 +1,195 @@
+* Entrypoint check - plugin registration: PASSED
+* Entrypoint check - registered modules exist: FAILED
+* Hooks registration check - napari-demo: PASSED
+* Hooks registration check - napari-else: FAILED
+----------------------------------------------------------------
+Error: 'Framework :: napari' does not exist for napari-demo's 10 known classifier(s)
+----------------------------------------------------------------
+Error: Did not find module napari_not_exists for plugin napari-else under package napari-demo
+----------------------------------------------------------------
+Error: No hook registered under plugin napari-else, Please see tutorial in https://napari.org/docs/dev/plugins/ to verify your plugin setup
+----------------------------------------------------------------
+```
+
+early termination of checks due to package having invalid entrypoint:
+```
+$ pip install napari-demo
+$ npd validate napari-demo
+----------------------------------------------------------------
+Scanning current python environment: /Library/Frameworks/Python.framework/Versions/3.8
+----------------------------------------------------------------
+Validating package napari-demo
+* Classifier check: PASSED
+* Entrypoint check - syntax: FAILED
+----------------------------------------------------------------
+Error: 'Framework :: napari' does not exist for napari-demo's 10 known classifier(s)
+----------------------------------------------------------------
+Error: Invalid entrypoint for package napari-demo, add 'napari.plugin' to the entrypoint under napari-demo
+----------------------------------------------------------------
+```
+```
+$ pip install napari-demo
+$ npd validate napari-demo
+----------------------------------------------------------------
+Scanning current python environment: /Library/Frameworks/Python.framework/Versions/3.8
+----------------------------------------------------------------
+Validating package napari-demo
+* Classifier check: PASSED
+* Entrypoint check - syntax: PASSED
+* Entrypoint check - plugin registration: FAILED
+* Entrypoint check - registered modules exist: FAILED
+----------------------------------------------------------------
+Error: No plugin registered for napari-demo, add plugin module registration under 'napari.plugin'
+----------------------------------------------------------------
+```
+
+---
+## npd discover
+
+### Description
+Discover plugins in current python environment, and show hook types for each plugin
+
+### Usage
+```
+$ npd discover
+```
+
+### Examples
+List plugins under current python environment:
+```
+$ pip install napari-demo
+$ pip install napari-svg
+$ pip install napari-hdf5-labels-io
+$ npd discover
+----------------------------------------------------------------
+Scanning current python environment: /Library/Frameworks/Python.framework/Versions/3.8
+----------------------------------------------------------------
+Found 1 registered function hook(s) for napari-demo:
```

Comment:
I would like to see a list of discovered plugins here, not including the hooks. If it's useful to also include hooks here, I would show them under the list or put them under a --verbose output. Also having the plugin name on a line above each "Found...hook(s) for..." would be easier to read.